modules/openapi-generator/src/main/resources/php/ObjectSerializer.mustache

```diff
@@ -140,7 +140,11 @@ class ObjectSerializer
      */
     public static function toHeaderValue($value)
     {
-       return json_encode(ObjectSerializer::sanitizeForSerialization($value));
+        if (method_exists($value, 'toHeaderValue')) {
+            return $value->toHeaderValue();
+        }
+
+        return self::toString($value);
```

Comment:
How about this instead of adding `toHeaderValue()` to `model_generic.mustache` ? 💡 

```php
// Strip newlines and spaces which comes from JSON_PRETTY_PRINT
return str_replace([PHP_EOL, ' '], '', (string)$value);
```

This will reduce the `if` condition.
>        if (method_exists($value, 'toHeaderValue')) {
>            return $value->toHeaderValue();
>        }

Reply:
There's no need to remove spaces, as they are allowed in a header value (just not at the beginning/end, but JSON wouldn't begin/end with spaces anyway). But yeah, otherwise this approach would work I suppose.

Reply:
I pushed this change. It is simpler, but now there are extra tabs in the JSON since it was pretty-printed, which doesn't really make sense for a single line. We could strip out the tabs as well, I suppose.

Reply:
Yes, that is why I removed spaces in the sample. 💡

Reply:
Yeah, I see. We should still allow spaces inside strings in the JSON. Would you prefer the previous solution then, with `toHeaderValue()` on the model, or keep it like this?

Reply:
Oh... sorry for my oversight... I would prefer the previous solution, which makes clean output.

Reply:
:+1: